expo/expo
# Add check in expo-doctor for expo-permissions

# Why



When upgrading to SDK50, I spent a while stuck on the following error message
Execution failed for task ':expo-permissions:compileReleaseKotlin'.
> 'compileReleaseJavaWithJavac' task (current target is 17) and 'compileReleaseKotlin' task (current target is 11) jvm target compatibility should be set to the same Java version.
Turns out we still had expo-permissions installed, even though we weren't using it any more (in a previous SDK upgrade we make the necessary refactor to do the permission requests from the relevant expo modules)
It'd be great if expo-doctor would warn about this issue. If you are on >=SDK50 and have expo-permissions, you should uninstall expo-permissions

# How


Added an extra test in expo-doctor

# Test Plan


- checkout project with expo sdk50
- install `expo-permissions`
- run `npx expo-doctor`
- check that you get an error
# Checklist
